packages/core/src/components/time-picker-rework/time-picker-rework.tsx

```diff
@@ -0,0 +1,337 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Siemens AG
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {
+  Component,
+  Event,
+  EventEmitter,
+  forceUpdate,
+  h,
+  Host,
+  Method,
+  Prop,
+  State,
+  Watch,
+} from '@stencil/core';
+import dayjs, { UnitType } from 'dayjs';
+import customParseFormat from 'dayjs/plugin/customParseFormat';
+import { DateTimeCardCorners } from '../date-time-card/date-time-card';
+
+export type TimePickerCorners = DateTimeCardCorners;
+
+dayjs.extend(customParseFormat);
+
+interface TimePickerDescriptor {
+  unit: 'hours' | 'minutes' | 'seconds';
+  placeholder: string;
+  hidden: boolean;
+}
+
+interface TimeOutputFormat {
+  hours: string;
+  minutes: string;
+  seconds: string;
+}
+
+/**
+ * @internal
+ */
+@Component({
+  tag: 'ix-time-picker-rework',
+  styleUrl: 'time-picker-rework.scss',
+  shadow: true,
+})
+export class TimePickerRework {
+  /**
+   * Format of time string
+   *
+   * @since 1.1.0
+   */
+  // @Prop() format: string = 'HH:mm:ss';
+  @Prop() format: string = 'HH:mm:ss';
+
+  /**
+   * Corner style
+   */
+  @Prop() corners: TimePickerCorners = 'rounded';
+
+  /**
+   * Controls the visual presentation and styling of the component when it is displayed as a standalone element
+   */
+  @Prop() standaloneAppearance: boolean = true;
+
+  /**
+   * Show hour input
+   */
+  @Prop() showHour = false;
+
+  /**
+   * Show minutes input
+   */
+  @Prop() showMinutes = false;
+
+  /**
+   * Show seconds input
+   */
+  @Prop() showSeconds = false;
+
+  /**
+   * Select time with format string
+   *
+   * @since 1.1.0
+   */
+  @Prop() time: string = dayjs().format(this.format);
+
+  @Watch('time')
+  watchTimePropHandler(newValue: string) {
+    this._time = dayjs(newValue, this.format, true);
+    if (!this._time.isValid()) {
+      throw new Error('Format is not supported or not correct');
+    }
+  }
+
+  /**
+   * Set time reference
+   */
+  @Prop() timeReference: 'AM' | 'PM' | undefined;
+
+  /**
+   * Text of date select button
+   *
+   * @since 1.1.0
+   */
+  @Prop() textSelectTime = 'Done';
+
+  /**
+   * Time event
+   */
+  @Event() done: EventEmitter<string>;
+
+  /**
+   * Time change event
+   */
+  @Event() timeChange: EventEmitter<string>;
```

Comment:
Time change event will be triggered twice:

```

@Component({
  tag: 'my-component',
  styleUrl: 'my-component.scss',
  scoped: true,
})
export class MyComponent {
  @State() timeValue = '12:12:12';

  render() {
    return (
      <Host>
        <ix-button
          onClick={() => {
            console.log('change');
            this.timeValue = '11:11:11';
          }}
        >
          Change time
        </ix-button>
        <ix-time-picker-rework
          time={this.timeValue}
          onTimeChange={console.log}
          onDone={console.log}
        ></ix-time-picker-rework>
      </Host>
    );
  }
}
```

Reply:
resolved